modules/apps/static/portal-lpkg-deployer/portal-lpkg-deployer-impl/src/main/java/com/liferay/portal/lpkg/deployer/internal/ContainerLPKGUtil.java

```diff
@@ -69,15 +63,20 @@ public static List<File> deploy(File lpkgFile, Properties properties)
 
 					return null;
 				}
+			}
+		}
 
-				Path lpkgPath = deployerDirPath.resolve(name);
+		Path deployerDirPath = Paths.get(
+			PropsValues.MODULE_FRAMEWORK_MARKETPLACE_DIR);
 
-				Files.copy(
-					zipFile.getInputStream(zipEntry), lpkgPath,
-					StandardCopyOption.REPLACE_EXISTING);
+		File deployerDirFile = deployerDirPath.toFile();
 
-				lpkgFiles.add(lpkgPath.toFile());
-			}
+		FileUtil.unzip(lpkgFile, deployerDirPath.toFile());
+
+		List<File> lpkgFiles = new ArrayList<>();
+
+		for (File file : deployerDirFile.listFiles()) {
```

Comment:
not sure if this is a correct way to get the file in the marketplace folder. If we already have files in the marketplace, then it will be also included in the lpkgFiles, which is wrong.